src/core/DataFile.cpp

```diff
@@ -1638,6 +1639,51 @@ void DataFile::upgrade_1_3_0()
 	}
 }
 
+void DataFile::upgrade_noHiddenAutomationTracks()
+{
+	// convert global automation tracks to non-hidden
+	QDomElement song = firstChildElement("lmms-project")
+		.firstChildElement("song");
+	QDomElement tc = song.firstChildElement("trackcontainer");
+	QDomElement gaTrack = song.firstChildElement("track");
+	if (!gaTrack.isNull()
+		&& gaTrack.attribute("type").toInt() == static_cast<int>(Track::Type::HiddenAutomation))
+	{
+		// global automationclips
+		QDomNodeList aps = gaTrack.elementsByTagName("automationclip");
+		QList<QDomNode> tracksToInsert;
+		for (int i = 0; i < aps.length(); ++i)
+		{
+			QDomElement ap = aps.item(i).toElement();
+			// If ap has time nodes, move it to trackcontainer
+			// There are times when an <object> node is present without an
+			// object with the same ID in the file, so we ignore that node
+			if (ap.elementsByTagName("time").length() > 1)
+			{
+				QDomElement aptrack = createElement("track");
+				aptrack.setAttribute("muted", QString::number(false));
```

Comment:
In general it's a bit unfortunate that the upgrade code must know the implementation details of how other entities represent their data in the XML file. However, that's a design problem and not something for this PR.